tests/integration/api_v2/rpc/delegates.test.js

```diff
@@ -0,0 +1,177 @@
+			const { result } = response;
+			expect(result).toMap(resultEnvelopeSchema);
+			expect(result.data).toBeInstanceOf(Array);
+			expect(result.data.length).toBeGreaterThanOrEqual(1);
+			result.data.forEach(account => {
+				expect(account).toMap(accountSchemaVersion5);
+				expect(account.username).toContain('genesis_1');
+			});
+			expect(result.meta).toMap(metaSchema, { offset: 0 });
+		});
+
+		it('returns one result when limit = 1', async () => {
+			const limit = 1;
+			const response = await getDelegates({ isDelegate: true, limit });
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result).toMap(resultEnvelopeSchema);
+			expect(result.data).toBeInstanceOf(Array);
+			expect(result.data.length).toEqual(limit);
+			result.data.forEach(account => {
+				expect(account).toMap(accountSchemaVersion5);
+				expect(account.dpos).toMap(dpos);
+			});
+			expect(result.meta).toMap(metaSchema, { count: limit, offset: 0 });
+		});
+	});
+
+	describe('returns delegates based on address', () => {
+		it('returns known delegate by address', async () => {
+			const response = await getDelegates({
+				isDelegate: true, address: refDelegate.summary.address,
+			});
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result).toMap(resultEnvelopeSchema);
+			expect(result.data).toBeArrayOfSize(1);
+			result.data.forEach(account => {
+				expect(account).toMap(accountSchemaVersion5);
+				expect(account.dpos).toMap(dpos);
+			});
+			expect(result.meta).toMap(metaSchema, { count: 1, offset: 0 });
+		});
+	});
+
+	describe('returns delegates based on username', () => {
+		it('returns known delegate by username', async () => {
+			const response = await getDelegates({
+				isDelegate: true, username: refDelegate.summary.username,
+			});
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result).toMap(resultEnvelopeSchema);
+			expect(result.data).toBeArrayOfSize(1);
+			result.data.forEach(account => {
+				expect(account).toMap(accountSchemaVersion5);
+				expect(account.dpos).toMap(dpos);
+			});
+			expect(result.meta).toMap(metaSchema, { count: 1, offset: 0 });
+		});
+	});
+
+	describe('returns delegates based on public key', () => {
+		it('returns known delegate by public key', async () => {
+			const response = await getDelegates({
+				isDelegate: true, publickey: refDelegate.summary.publicKey,
+			});
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result).toMap(resultEnvelopeSchema);
+			expect(result.data).toBeArrayOfSize(1);
+			result.data.forEach(account => {
+				expect(account).toMap(accountSchemaVersion5);
+				expect(account.dpos).toMap(dpos);
+			});
+			expect(result.meta).toMap(metaSchema, { count: 1, offset: 0 });
+		});
+	});
+
+	// Fix delegate search by status
+	xdescribe('returns delegates based on status', () => {
```

Comment:
Please refer to my earlier comments.

```suggestion
	describe('returns delegates based on status', () => {
```

Reply:
Same as above.